robolectric/src/main/java/org/robolectric/android/internal/AndroidTestEnvironment.java

```diff
@@ -682,16 +688,56 @@ private String createTempDir(String name) {
         .toString();
   }
 
+  private static BroadcastReceiver newBroadcastReceiverFromP(
+      String receiverClassName,
+      Application application,
+      LoadedApk loadedApk,
+      BroadcastReceiverData receiver) {
+    ClassLoader classLoader;
+    if (application.getBaseContext() != null) {
+      classLoader = application.getBaseContext().getClassLoader();
+    } else {
+      classLoader = loadedApk.getClassLoader();
+    }
+    for (String action : receiver.getActions()) {
```

Comment:
One issue that I am seeing is  it is possible that receiver.getActions() is an empty list. Previously this would still get that receiver to be registered, but now it will not get registered.

Reply:
I think I can try to fix this issue this weekend.

Reply:
Sounds good, I'll make this change and see if there are any other issues.

Reply:
But yeah it's possible to define a broadcast receiver in a manifest without any intent filters, and then send an intent to the broadcast receiver class directly.

Reply:
Yep. It can be a test case for non-action receiver.